src/main/java/org/lsc/jndi/JndiServices.java

```diff
@@ -557,9 +557,9 @@ private SearchResult doGetEntry(final String base, final String filter,
 		try {
 			sc.setSearchScope(scope);
 			String rewrittenBase = null;
-			if (contextDn != null && searchBase.toLowerCase().endsWith(contextDn.toString().toLowerCase())) {
-				if (!searchBase.equalsIgnoreCase(contextDn.toString())) {
-				rewrittenBase = searchBase.substring(0, searchBase.toLowerCase().lastIndexOf(contextDn.toString().toLowerCase()) - 1);
+			if (contextDn != null && searchBase.toLowerCase().endsWith(getContextDn().toLowerCase())) {
```

Comment:
here there is first a check that contextDn is not null, so the change on this line seems useless

Reply:
Yes, you are right! Thanks.